src/osdatahub/NamesAPI/names_api.py

```diff
@@ -0,0 +1,160 @@
+from collections.abc import Iterable
+from typing import Union
+
+import requests
+from geojson import FeatureCollection
+from typeguard import typechecked
+
+from osdatahub.NamesAPI.local_types import validate_local_type, get_local_type
+from osdatahub.extent import Extent
+from osdatahub.grow_list import GrowList
+from osdatahub.utils import addresses_to_geojson
+
+
+class NamesAPI:
+    """Main class for querying the OS Names API (https://osdatahub.os.uk/docs/names/overview)
+
+    Args:
+        key (str): A valid OS API Key. Get a free key here - https://osdatahub.os.uk/
+
+    Example::
+
+        from osdatahub import FeaturesAPI, Extent
+        from os import environ
+
+        key = environ.get("OS_API_KEY")
+        names = NamesAPI(key)
+        results = features.find("Buckingham Palace", limit=5)
+    """
+    __ENDPOINT = r"https://api.os.uk/search/names/v1/"
+    HEADERS = {"method": "POST",
+               "headers": "{'Content-Type': 'application/json'}"}
+
+    def __init__(self, key: str):
+        self.key = key
+
+    def __endpoint(self, api_name: str) -> str:
+        return self.__ENDPOINT + api_name + f"?key={self.key}"
+
+    @typechecked
+    def find(self, text: str, limit: int = 100,
+             bounds: Extent = None, bbox_filter: Extent = None,
+             local_type: Union[Iterable, str] = None) -> FeatureCollection:
+        """A free text query of the OS Names API
+
+        Args:
+            text (str): The free text search parameter
+            limit (int, optional): The maximum number of features to return.
+                Defaults to 100.
+            bounds (Extent, optional): Biases the results to a certain area. Must be British National Grid
+                (EPSG:27700) CRS
+            bbox_filter (Extent, optional): Filters the results to a certain area. Must be British National Grid
+                (EPSG:27700) CRS
+            local_type (Union[Iterable, str], optional): Filters the results to certain local types. Available local
+                types can be found at the bottom of https://osdatahub.os.uk/docs/names/technicalSpecification
+
+        Returns:
+            FeatureCollection: The results of the query in GeoJSON format
+        """
+        data = GrowList()
+        params = {"query": text}
+
+        if limit <= 0:
+            raise ValueError("Parameter \"limit\" must be a positive integer")
+
+        if bounds:
+            if not bounds.crs == "EPSG:27700":
```

Comment:
Not sure if this is more pythonic
```suggestion
            if bounds.crs != "EPSG:27700":
```